spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileUtils.java

```diff
@@ -76,4 +76,14 @@ private RemoteFileUtils() {}
 		}
 	}
 
+	/**
+	 * Invoke this method to clean up and close the session after using {@link Session#readRaw(String)}.
+	 * @param session The session.
+	 * @throws IOException if an IO error occurs.
+	 */
+	public static <F> void closeSession(Session<F> session) throws IOException {
+		session.finalizeRaw();
+		session.close();
```

Comment:
Good point - the `close()` can test the `readingRaw` field. Eliminating the utility method; the SpEL would simply be 

    headers['file_remoteSession'].close()